xchange-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenOrderBookUtils.java

```diff
@@ -97,4 +92,26 @@ private static long timestampToMs(String timestamp) {
         return new BigDecimal(timestamp).multiply(new BigDecimal(1000)).longValue();
     }
 
+    public static OrderBook verifyKrakenOrderBook(OrderBook orderBook) {
```

Comment:
General points on design:

This method has a functional-style signature but has side effects.

`Collections.sort` and `removeIf` both mutate the existing lists, so you are doing multiple things:

1. Sorting the _existing_ order book
1. Modifying the_ existing_ order book to remove entries
1. Creating a new order book with the modified entry lists (leaving two identical `OrderBook` objects containing the same entries).

Instead, it would make more sense to either create the new `Orderbook` using a filtered copy and sorted of the original (using a consistent sort) OR return `void` and explicitly modify the existing order book.

The method is called `verify...` so I'd usually expect it to check something.  Would `sortAndRemoveInvalidEntries` make more sense?

In any case, I think this code is in the wrong place. It should happen in `adaptOrderBook` _before_ creating the `OrderBook`.

I'm still a little unclear on what issue this is resolving, too.

Reply:
OK, i would just return a void. The issue that this method fix is that a lot of times the stream doesn't send some messages and there are times that the orderbook is not consistent with the real Kraken orderbook. I use this implementation on my own project and it works very well. I have found the same issue on CoinbaseProOrderbook also and i will try to fix this too.

Reply:
I believe that there is a general issue with inconsistent orderbooks in this library, i have encounter issues on Bitmex,Kraken,CoinbasePro exchange. I don't use any REST API calls to get a new snapshot. As i have seen in issue #387 , there are more exchanges with this inconsistent orderbook thing. Maybe we need to discuss about implement a global approach in this, maybe we should combine a websocket stream with REST API, or at least create a validation method in order to delete all the inconsistent orderbook entries, like the verifyOrderbook method.